localstack/services/cloudformation/models/ssm.py

```diff
@@ -42,17 +43,43 @@ def update_resource(self, new_resource, stack_name, resources):
         ]
         update_config_props = select_attributes(props, parameters_to_select)
 
-        if "Tags" in update_config_props:
-            update_config_props["Tags"] = [
-                {
-                    "Key": k,
-                    "Value": v,
-                }
-                for (k, v) in (update_config_props["Tags"] or {}).items()
-            ]
+        ssm_client = connect_to().ssm
 
-        client = aws_stack.connect_to_service("ssm")
-        return client.put_parameter(Overwrite=True, **update_config_props)
+        # tag handling
+        new_tags = update_config_props.pop("Tags", {})
+        current_tags = ssm_client.list_tags_for_resource(
+            ResourceType="Parameter", ResourceId=self.props.get("Name")
+        )["TagList"]
+        current_tags = {tag["Key"]: tag["Value"] for tag in current_tags}
+
+        new_tag_keys = set(new_tags.keys())
+        old_tag_keys = set(current_tags.keys())
+        potentially_modified_tag_keys = new_tag_keys.intersection(old_tag_keys)
+        tag_keys_to_add = new_tag_keys.difference(old_tag_keys)
+        tag_keys_to_remove = old_tag_keys.difference(new_tag_keys)
+
+        for tag_key in potentially_modified_tag_keys:
+            # also overwrite changed tags
+            if new_tags[tag_key] != current_tags[tag_key]:
+                tag_keys_to_add.add(tag_key)
+
+        if tag_keys_to_add:
+            ssm_client.add_tags_to_resource(
+                ResourceType="Parameter",
+                ResourceId=self.props.get("Name"),
+                Tags=[
+                    {"Key": tag_key, "Value": tag_value}
+                    for tag_key, tag_value in new_tags.items()
+                    if tag_key in tag_keys_to_add
+                ],
+            )
+
+        if tag_keys_to_remove:
+            ssm_client.remove_tags_from_resource(
+                ResourceType="Parameter", ResourceId=self.props.get("Name"), Tags=tag_keys_to_remove
+            )
```

Comment:
this part isn't actually working yet because the engine passes the old tag state for some reason when updating :shrug: 

Will be fixed in the course of the next CFn core projects